indico/modules/events/registration/clone.py

```diff
@@ -141,6 +143,15 @@ def _clone_registrations(self, old_form, new_form, field_data_map):
                 if old_registration_data.storage_file_id is not None:
                     with old_registration_data.open() as fd:
                         new_registration_data.save(fd)
+                # Assigns the mapped session blocks for cloned event to the cloned registration.
+                if new_registration_data.field_data.field.input_type == 'sessions':
+                    mapped_sessions_data = []
+                    # If timetable is not cloned, assign an empty array to this field.
+                    if session_blocks_map:
+                        mapped_sessions_data = [next((session_blocks_map['session'].id
+                                                for session in session_blocks_map if session.id == _id), None)
+                                                for _id in new_registration_data.data]
+                    new_registration_data.data = mapped_sessions_data
```

Comment:
Do we need to filter out `None` values where? I think they can happen when a session block has been deleted but someone had selected it.

Reply:
That was indeed the case. Fixed now.